tests/testsuite/publish.rs

```diff
@@ -1688,46 +1733,106 @@ fn in_workspace() {
             "#,
         )
         .file("foo/src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("publish --no-verify --token sekrit -p foo")
+        .with_status(101)
+        .with_stderr("error: can't use \"--package <SPEC>\" in virtual manifest")
+        .run();
+}
+
+#[cargo_test]
+fn in_package_workspace_not_found() {
+    registry::init();
+
+    let p = project()
         .file(
-            "bar/Cargo.toml",
+            "Cargo.toml",
             r#"
-                [project]
-                name = "bar"
+                [package]
+                name = "foo"
+                version = "0.1.0"
+                edition = "2018"
+                [workspace]
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .file(
+            "li/Cargo.toml",
+            r#"
+                [package]
+                name = "li"
                 version = "0.0.1"
+                edition = "2021"
                 authors = []
                 license = "MIT"
-                description = "bar"
-                workspace = ".."
+                description = "li"
             "#,
         )
-        .file("bar/src/main.rs", "fn main() {}")
+        .file("li/src/main.rs", "fn main() {}")
         .build();
 
-    p.cargo("publish --no-verify --token sekrit -p foo")
+    p.cargo("publish -p li --no-verify --token sekrit ")
+        .with_status(101)
         .with_stderr(
             "\
-[UPDATING] [..]
-[WARNING] manifest has no documentation, [..]
-See [..]
-[PACKAGING] foo v0.0.1 ([CWD]/foo)
-[UPLOADING] foo v0.0.1 ([CWD]/foo)
+error: not found `li` in manifest members. Check in manifest path `[CWD]`
 ",
         )
         .run();
+}
 
-    validate_upload_foo();
+#[cargo_test]
+fn in_package_workspace_found_mutilate() {
```

Comment:
Is this maybe a typo?

```suggestion
fn in_package_workspace_found_multiple() {
```